src/app/(main)/product/[id]/page.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
```

Comment:
You want to avoid using `use client` in `page.tsx` components as much as possible. If a logic requires that, put that but into a sub-component